src/libratbag-data.c

```diff
@@ -504,14 +540,14 @@ ratbag_device_data_new_for_id(struct ratbag *ratbag, const struct input_id *id)
 			goto out;
 	}
 
-	if (!data)
+	if (data == NULL)
```

Comment:
haven't tested this, so can you explain why we need to change this?

Reply:
Sorry, force of habit. We don't need to change this because it does the same thing, but I find it more expressive.
If i see ! somewhere I think of booleans, but this is a pointer.

I can change it back if you want.

Reply:
Either that, or do it in separate commit. Just keep the commit history clear, so that we can trace back any changes properly.